mcs/class/Mono.Posix/Mono.Unix.Native/Syscall.cs

```diff
@@ -727,6 +727,194 @@ public enum MremapFlags : ulong {
 		MREMAP_MAYMOVE = 0x1,
 	}
 
+	[Map]
+	[CLSCompliant (false)]
+	public enum SockType : int {
```

Comment:
Indeed, that's a brilliant search, and if we [extend it to include `SocketType`](https://www.google.com/webhp#q=%22using+system.net.sockets%22+%22using+mono.unix.native%22+SocketType), there are *10* matches.

Which isn't a high number, though some of those collisions also show that [e.g. `AddressFamily` will cause issues](http://casa-auth.googlecode.com/svn/trunk/CASA/micasad/communication/UnixCommunication.cs):

            listeningSocket = new Socket( AddressFamily.Unix,
                                          SocketType.Stream,
                                          ProtocolType.IP );

This suggests that `SocketType`/`AddressFamily`/etc. shouldn't be used (`:-(`), but what about a `Unix` prefix, e.g. `UnixSocketType`, `UnixAddressFamily`, etc.? Does that sound reasonable?